socorro/external/es/crashstorage.py

```diff
@@ -319,6 +320,27 @@ def capture_crash_metrics(self, raw_crash, processed_crash):
             # later.
             self.config.logger.exception('something went wrong when capturing processed_crash_size')
 
+    def _index_crash_in_elasticsearch(self, connection, es_index, es_doctype, crash_document,
+                                      crash_id):
+        try:
+            start_time = datetime.now()
+            connection.index(
+                index=es_index,
+                doc_type=es_doctype,
+                body=crash_document,
+                id=crash_id
+            )
+            index_outcome = 'successful'
+        except Exception:
+            index_outcome = 'failed'
+            raise
+        finally:
+            elapsed_time = datetime.now() - start_time
+            self.metrics.histogram(
+                '%s_index' % index_outcome,
+                value=elapsed_time.microseconds
```

Comment:
There are two problems here. First is that the `.microseconds` property is not the total number of microseconds in the timedelta, but just the microseconds part of the timespan. So if the timedelta was 1.01, then this would just be .01 which isn't what we want.

```
$ python
Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from datetime import datetime
>>> start = datetime.now()
>>> delta = datetime.now() - start
>>> delta
datetime.timedelta(0, 6, 479467)
>>> delta.microseconds
479467
```

https://docs.python.org/2/library/datetime.html#datetime.timedelta

Second is that we want timings in milliseconds--not microseconds. That keeps our timings in datadog consistent with the same metric.

When I implemented this in Markus, I did it with `time.time()` which gives you more of what you want without having to do any math or conversions.

This is what Markus does:

https://github.com/willkg/markus/blob/master/markus/main.py#L339-L342

We should change it to something like that.

Reply:
Oh, I see what happened--you copied the code from the previous attempt. That's totally fine. However, what's funny about that is that @sciurus and I puzzled over the stats a couple of months ago when I rewrote the metrics specifically about the milliseconds vs. microseconds and how the crash storage timings prior to the rewrite were wrong (for the last 6+ years). I should have tossed a comment in his PR back then, but missed it. Sorry about that!